dashboard/src/main/java/com/google/cloud/tools/opensource/cloudbomdashboard/ArtifactMavenData.java

```diff
@@ -220,20 +233,31 @@ private static String getSharedDependenciesVersionFromUrl(String pomUrl) {
       FileUtils.copyInputStreamToFile(input, pomFile);
       MavenXpp3Reader read = new MavenXpp3Reader();
       Model model = read.read(new FileInputStream(pomFile));
-      if (model.getDependencyManagement() == null) {
-        return null;
-      }
-      for (Dependency dep : model.getDependencyManagement().getDependencies()) {
-        if ("com.google.cloud".equals(dep.getGroupId())
-            && "google-cloud-shared-dependencies".equals(dep.getArtifactId())) {
-          if (dep.getVersion().startsWith("${")) {
-            String sharedVersion = dep.getVersion().substring(1).replaceAll("[{}]", "");
-            return model.getProperties().getProperty(sharedVersion);
-          }
-          return dep.getVersion();
-        }
-      }
-
+      ModelBuildingRequest request = new DefaultModelBuildingRequest();
+      request.setRawModel(model);
+      RepositorySystem repositorySystem = RepositoryUtility.newRepositorySystem();
+      RepositorySystemSession session = RepositoryUtility.newSession(repositorySystem);
+
+      // Set model resolver to locate parent POM on Maven Central.
+      RemoteRepository mavenCentral =
+          new RemoteRepository.Builder("central", "default", "https://repo1.maven.org/maven2/")
+              .build();
+      request.setModelResolver(
+          new ProjectModelResolver(
+              session,
+              null,
+              repositorySystem,
+              new DefaultRemoteRepositoryManager(),
+              ImmutableList.of(mavenCentral),
+              null,
+              null));
+
+      // Capture java version from system environment for profile activation.
+      request.setSystemProperties(System.getProperties());
+
+      ModelBuilder builder = new DefaultModelBuilderFactory().newInstance();
+      Model effectiveModel = builder.build(request).getEffectiveModel();
+      return effectiveModel.getProperties().getProperty("google-cloud-shared-dependencies.version");
     } catch (XmlPullParserException | IOException ignored) {
```

Comment:
Do we know why these exceptions are ignored, but the ModelBuildingException is thrown?

Reply:
That is a good question. It is a little unclear as to why the exceptions were ignored.  https://effectivejava.googleplex.com/third-edition#page=331 advices against this especially without appropriate comments explaining the reasoning and suggests the following to avoid silent failures:

```
Merely letting an exception propagate outward can at least cause the
program to fail swiftly, preserving information to aid in debugging the failure.
```